### PR TITLE
video meta for the video bar

### PR DESCRIPTION
**Pull request by Alqanar**

### CODE REVIEW

**Review by iAdramelk — APPROVED** [review]
Code is ok, with couple nitpiks, but small mobile version looks not very good with 4 client in  one row. Maybe make it 2 instead of 4?

Also I think line height it title on mobile is too large.

Can you discuss it with @alexwolfe?

File: utils/youtube-meta.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+const addZero = (num: number) => String(num).padStart(2, `0`);
+
+export async function fetchVideoMeta(id: string): Promise<Meta | FullMeta> {
+  let data: Meta | FullMeta = {
+    href: `${YOUTUBE_URL}?v=${id}`,
+  };
+
+  if (!NEXT_PUBLIC_YOUTUBE_API_KEY && process.env.NODE_ENV === "development") {
```
Comment (iAdramelk) [inline utils/youtube-meta.ts:71]: Nit: Let's remove `process.env.NODE_ENV === "development"`. We may want to run local build to test some stuff without the key too.

File: utils/youtube-meta.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    data = {
+      href: "#",
+      title: "This is a fake video link for development mode",
+      thumbnail:
+        "https://i.ytimg.com/vi/A93bGm4I2a4/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCNpZMhxXTaUdaTDBjm3aFZvjJD3Q",
```
Comment (iAdramelk) [inline utils/youtube-meta.ts:76]: Nit: Let's use local placeholder image here. Any existing video on youtube may be deleted at any point of time with its thumbnails.

File: utils/youtube-meta.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+interface RawItem {
+  contentDetails: RawContentDetails;
+  snippet: RawSnippet;
+}
+
+interface RawVideoMeta {
+  items: RawItem[];
+}
+
+const addZero = (num: number) => String(num).padStart(2, `0`);
```
Comment (iAdramelk) [inline utils/youtube-meta.ts:64]: Nit: Let's move it before `changeFormatDuration`.

File: utils/remark-import-frontmatter.ts [hunk 4]
```diff
@@ -20,16 +22,20 @@ interface RemarkImportFrontmatterOptions {
...
     const firstChild = children[0];
-    let config = {};
+    let config: { videoBanner?: string | Meta | FullMeta } = {};
 
     if (firstChild && firstChild.type === "yaml") {
       config = yaml.load(firstChild.value as string) as Record<string, unknown>;
     }
 
-    root.children.unshift(createMdxjsEsmNode(name, config));
+    if ("videoBanner" in config && typeof config.videoBanner === "string") {
```
Comment (iAdramelk) [inline utils/remark-import-frontmatter.ts:35]: Let's add an option to the plugin that will allow us to disable video fetching and remove `videoBanner` from the config.

This way, if we don't have internet locally or have some problems with the hosting we can temporarily disable banners.

**Review by iAdramelk — COMMENTED** [review]
Good work! Just a few small nitpics.

**Review by iAdramelk — APPROVED** [review]

**Comment by Alqanar** [comment]
https://github.com/gravitational/next/issues/322 https://github.com/gravitational/next/issues/222
Added video bar component.
To use this component, have to pass to Front Matter the "videoBanner" key with the YouTube video identifier in .mdx file. The ID is stored in the URL in the part that follows the “?v=...” characters.

**Pull request merged**